lib/class-dsl.php

```diff
@@ -158,16 +158,23 @@ public static function all_terms( string $field, array $values ): array {
 	 * Given DateTime objects for the from and to components of a range query,
 	 * builds the from/to components of the range query and returns it.
 	 *
-	 * @param DateTime $from The start date.
-	 * @param DateTime $to   The end date.
+	 * @param string|DateTime $from The start date.
+	 * @param string|DateTime $to   The end date.
 	 *
 	 * @return array An array containing a from and to key, usable in a range query.
 	 */
-	public function build_range( DateTime $from, DateTime $to ): array {
-		return [
-			'from' => $from->format( 'Y-m-d H:i:s' ),
-			'to'   => $to->format( 'Y-m-d H:i:s' ),
-		];
+	public function build_range( $from, $to ): array {
+		$range = [];
+
+		if ( ! empty( $from ) && $from instanceof DateTime ) {
+			$range['from'] = $from->format( 'Y-m-d H:i:s' );
+		}
+
+		if ( ! empty( $to ) && $to instanceof DateTime ) {
```

Comment:
These empty checks aren't bringing anything to the table

```suggestion
		if ( $from instanceof DateTime ) {
			$range['from'] = $from->format( 'Y-m-d H:i:s' );
		}

		if ( $to instanceof DateTime ) {
```